server/src/test/java/org/elasticsearch/gateway/MetaDataStateFormatTests.java

```diff
@@ -291,6 +292,96 @@ public void testLoadState() throws IOException {
         }
     }
 
+    private DummyState writeAndReadStateSuccessfully(Format format, AtomicBoolean injectFailures, Path... paths) throws IOException {
+        injectFailures.set(false);
+        DummyState state = new DummyState(randomRealisticUnicodeOfCodepointLengthBetween(1, 100), randomInt(), randomLong(), randomDouble(), randomBoolean());
+        format.write(state, paths);
+        assertEquals(state, format.loadLatestState(logger, NamedXContentRegistry.EMPTY, paths));
+        return state;
```

Comment:
can we check that all other files have been properly cleaned up?

Reply:
++